### DISCUSSION TITLE
Add yarpidl_rosmsg source files to my code or to a yarp library

### DISCUSSION THREAD

**Discussion by plinioMoreno** [robotology/community#79] [Q&A] [answered]
I want to use the classes defined in [rosmsg source](https://github.com/robotology/yarp/tree/master/src/idls/rosmsg/src) in order to fill the type of a message.

Using cmake and the variables defined by YARP cmake files, how should I add those files to my code?

It would be possible to add those files to a YARP library?

cc @drdanz

**Comment by Tobias-Fischer** [comment]
Hi,
did you check out http://www.yarp.it/rostypes_tutorial_portable.html?

Best, Tobias

**Comment by plinioMoreno** [comment]
@Tobias-Fischer, the tutorial does not fill type anywhere. In the tutorial they are executing the yarpidl_rosmsg externally. Instead of doing that, I want to write a code that given a ros message name, it is able to fill the type [void Port::promiseType   ( const Type &  typ) ](http://www.yarp.it/classyarp_1_1os_1_1Port.html#a188b9971b46203b10b2ed9e6b58f886f) in my code.

There is a class that deals with that, which is located in the [rosmsg source](https://github.com/robotology/yarp/tree/master/src/idls/rosmsg/src). And I would like to use it in my code.

Plinio

**Comment by drdanz** [comment, marked as answer]
@plinioMoreno If I understand correctly your issue, I _think_ that you can do something like `port.promiseType(Type::byNameOnWire("your_package/your_message"));`

**Comment by barbalberto** [comment]
I'm not sure to understand your issue, can you be more precise on what you are trying to achieve?
Do you want to have a yarp port able to communicate with a ros topic using a specific ros message, and do everything from code?

**Comment by plinioMoreno** [comment]
@barbalberto, yes, I do. @drdanz, your guess was correct and works properly.

The reason for writing code like this is to get rid of issues on the [yarp-bottle-generator](https://github.com/vislab-tecnico-lisboa/yarp-bottle-generator), that we use to generate bridging code between ROS and YARP.

Thanks

**Comment by barbalberto** [comment]
So the best way to do it is to use the Publisher, which is a port already customized for usage with ROS, like the controlBoardWrapper is doing.

`yarp::os::Publisher<my_idl_generated_message>         rosPublisherPort;           // Dedicated ROS topic publisher`
